freqtrade/exchange/exchange.py

```diff
@@ -1299,6 +1304,27 @@ def fetch_positions(self) -> List[Dict]:
         except ccxt.BaseError as e:
             raise OperationalException(e) from e
 
+    @retrier
+    def fetch_trading_fees(self) -> Dict[str, Any]:
+        """
+        Fetch user account trading fees
+        Can be cached, should not update often.
+        """
+        if (self._config['dry_run'] or self.trading_mode != TradingMode.FUTURES
```

Comment:
why only futures?

Reply:
because fee detection from "trades" works in spot markets.

using `fetch_my_trades()` i can get the appropriate "fee" entry (e.g.:)
That's not working for futures, as there the `fetch_my_trades` endpoint doesn't contain fees.

```
[{'info': {'id': '3011005184',
   'create_time': '1647285410',
   'create_time_ms': '1647285410867.203000',
   'currency_pair': 'SOL_USDT',
   'side': 'sell',
   'role': 'maker',
   'amount': '0.25',
   'price': '79.1649',
   'order_id': '132746780837',
   'fee': '0',
   'fee_currency': 'USDT',
   'point_fee': '0.03958245',
   'gt_fee': '0'},
  'id': '3011005184',
  'timestamp': 1647285410867,
  'datetime': '2022-03-14T19:16:50.867Z',
  'symbol': 'SOL/USDT',
  'order': '132746780837',
  'type': None,
  'side': 'sell',
  'takerOrMaker': 'maker',
  'price': 79.1649,
  'amount': 0.25,
  'cost': 19.791225,
  'fee': {'cost': 0.0, 'currency': 'USDT'},      <<< we will read this
  'fees': [{'currency': 'USDT', 'cost': 0.0}]}]
```

Reply:
> because fee detection from "trades" works in spot markets.
> 
> using `fetch_my_trades()` i can get the appropriate "fee" entry (e.g.:) That's not working for futures, as there the `fetch_my_trades` endpoint doesn't contain fees.
> 
> ```
> [{'info': {'id': '3011005184',
>    'create_time': '1647285410',
>    'create_time_ms': '1647285410867.203000',
>    'currency_pair': 'SOL_USDT',
>    'side': 'sell',
>    'role': 'maker',
>    'amount': '0.25',
>    'price': '79.1649',
>    'order_id': '132746780837',
>    'fee': '0',
>    'fee_currency': 'USDT',
>    'point_fee': '0.03958245',
>    'gt_fee': '0'},
>   'id': '3011005184',
>   'timestamp': 1647285410867,
>   'datetime': '2022-03-14T19:16:50.867Z',
>   'symbol': 'SOL/USDT',
>   'order': '132746780837',
>   'type': None,
>   'side': 'sell',
>   'takerOrMaker': 'maker',
>   'price': 79.1649,
>   'amount': 0.25,
>   'cost': 19.791225,
>   'fee': {'cost': 0.0, 'currency': 'USDT'},      <<< we will read this
>   'fees': [{'currency': 'USDT', 'cost': 0.0}]}]
> ```

But this is a general exchange method, not a gateio method, so maybe for other exchanges this isn't correct

Reply:
well most exchanges provide fee as part of the (filled) order response.

we only ever query `fetch_my_trades` for exchanges that don't (binance, gateio futures) that i know off.
until now - we fees detected from here were working fine - and i'll worry about a wrong fee case once i encounter it ;)